src/Inventory/Asset/NetworkEquipment.php

```diff
@@ -137,7 +137,7 @@ public function prepare(): array
                 $stack->model = $switch->model;
                 $stack->$model_field = $switch->model;
                 $stack->description = $stack->name . ' - ' . ($switch->name ?? $switch->description);
-                $stack->name = $stack->name . ' - ' . ($switch->name ?? $switch->description);
+                $stack->name = $stack->name . '_' . $switch->stack_number;
```

Comment:
```suggestion
                $stack->name = $stack->name . ' - ' . ($switch->name ?? $switch->description) . ' - ' . $switch->stack_number;
```

Keep old format because the device name is relevant (and it's needed by https://github.com/glpi-project/glpi/pull/14187)  and add the stack number at the end

Result with your file

![image](https://github.com/glpi-project/glpi/assets/7335054/ead2cd53-49c9-424b-9cf0-423f7360e23b)